src/ServiceControl/Alerting/Mail/CustomChecksMailNotification.cs

```diff
@@ -0,0 +1,61 @@
+﻿namespace ServiceControl.Alerting.Mail
+{
+    using System.Threading.Tasks;
+    using Alerting;
+    using Contracts.CustomChecks;
+    using Infrastructure.DomainEvents;
+    using Raven.Client;
+    using ServiceBus.Management.Infrastructure.Settings;
+
+    class CustomChecksMailNotification : IDomainHandler<CustomCheckFailed>, IDomainHandler<CustomCheckSucceeded>
```

Comment:
We might want to be a little defensive in these handlers. Domain event handlers are not isolated from each other and an exception in one will bubble up to where the domain event was raised. If the SMTP server is down, this will disrupt the business transaction that persists the custom check state, and may prevent the event from being published to the transport.

Reply:
They are also run synchronously so the transaction will be delayed if the email server is unreachable.

Reply:
I've added exception handling and logging around SMTP calls. The send operation is now scheduled as an asynchronous async operation that will not delay the execution of other domain event handlers

Reply:
After more consideration, we went with [this](https://github.com/Particular/ServiceControl/pull/2501#issue-614492658) design. I think it's a good enough compromise.